lib/fs.js

```diff
@@ -609,6 +609,12 @@ fs.read = function(fd, buffer, offset, length, position, callback) {
     };
   }
 
+  if (length === 0) {
+    return process.nextTick(function() {
```

Comment:
Hmmm, this is not right. There is actually a deprecated `fs.read` form which returns `string`.

So,

1. This will actually return a buffer and the deprecated `fs.read` will collapse.
2. The order of parameters in string form is actually string and the bytesRead.